server/chat-plugins/random-battles.ts

```diff
@@ -0,0 +1,391 @@
+			if (!set.level) buf += `<li>Level: 50</li>`;
+			if (set.level && set.level < 50) buf += `<li>Level: ${set.level}</li>`;
+			if (set.shiny) buf += `<li>Shiny: Yes</li>`;
+			if (set.happiness) buf += `<li>Happiness: ${set.happiness}</li>`;
+			if (set.evs) {
+				buf += `<li>EVs: `;
+				const evs: string[] = [];
+				let ev: string;
+				for (ev in set.evs) {
+					if (set.evs[ev] === 0) continue;
+					evs.push(`${set.evs[ev]} ${statNames[ev]}`);
+				}
+				buf += `${evs.join(" / ")}</li>`;
+			}
+			buf += `<li>${Array.isArray(set.nature) ? set.nature.map(formatNature).join(" / ") : formatNature(set.nature)} Nature</li>`;
+			if (set.ivs) {
+				buf += `<li>IVs: `;
+				const ivs: string[] = [];
+				let iv: string;
+				for (iv in set.ivs) {
+					if (set.ivs[iv] === 31) continue;
+					ivs.push(`${set.ivs[iv]} ${statNames[iv]}`);
+				}
+				buf += `${ivs.join(" / ")}</li>`;
+			}
+			for (const moveid of set.moves) {
+				buf += `<li>- ${Array.isArray(moveid) ? moveid.map(formatMove).join(" / ") : formatMove(moveid)}</li>`;
+			}
+			buf += `</ul></details>`;
+		}
+	}
+	return buf;
+}
+
+export const commands: ChatCommands = {
+	'!randombattles': true,
+	randbats: 'randombattles',
+	randombattles(target, room, user) {
+		if (!this.runBroadcast()) return;
+		const args = target.split(',');
+		if (!args[0]) return this.parse(`/help randombattles`);
+		let dex = Dex;
+		if (args[1] && toID(args[1]) in Dex.dexes) dex = Dex.dexes[toID(args[1])];
+		const template = dex.getTemplate(args[0]);
+		if (!template.exists) {
+			return this.errorReply(`Error: Pok\u00e9mon '${args[0].trim()}' does not exist.`);
+		}
+		const formatName = dex.getFormat(`gen${dex.gen}randombattle`).name;
+		if (toID(args[1]) === 'gen1') {
+			const rbyMoves = getRBYMoves(template);
+			if (!rbyMoves) {
+				return this.errorReply(`${template.species} has no Random Battle data in ${GEN_NAMES[toID(args[1])]}`);
+			}
+			return this.sendReplyBox(`<span style="color:#999999;">Moves for ${template.species} in ${formatName}:</span><br />${rbyMoves}`);
+		}
+		if (toID(args[1]) === 'gen2') {
+			const gscMoves = getGSCMoves(template);
+			if (!gscMoves) {
+				return this.errorReply(`${template.species} has no Random Battle data in ${GEN_NAMES[toID(args[1])]}`);
+			}
+			return this.sendReplyBox(`<span style="color:#999999;">Moves for ${template.species} in ${formatName}:</span><br />${gscMoves}`);
+		}
+		if (!template.randomBattleMoves) {
+			return this.errorReply(`Error: No moves data found for ${template.species}${`gen${dex.gen}` in GEN_NAMES ? ` in ${GEN_NAMES[`gen${dex.gen}`]}` : ``}.`);
+		}
+		const moves = template.randomBattleMoves.map(formatMove);
+		this.sendReplyBox(`<span style="color:#999999;">Moves for ${template.species} in ${formatName}:</span><br />${moves.join(`, `)}`);
+	},
+	randombattleshelp: [
+		`/randombattles OR /randbats [pokemon], [gen] - Displays a Pok\u00e9mon's Random Battle Moves. Defaults to Gen 8.`,
```

Comment:
Shouldn't it default to the current battle's gen in a battle room?

Reply:
Good point; will add soon

Reply:
did